<?xml version="1.0"?>
<launch>
<arg name="fake" default="true"/>
<arg name="use_rviz" default="true" />
<arg name="debug" default="false" />
<arg name="pipeline" default="ompl" />
<arg name="gazebo" default="false" />

<arg unless="$(arg debug)" name="launch_prefix" value="" />
<arg     if="$(arg debug)" name="launch_prefix"
 value="gdb --ex run --args" />

<!-- load URDF&co -->
 <include file="$(find nj_220_2_7_moveit_config)/launch/planning_context.launch">
   <arg name="load_robot_description" value="false"  if     = "$(arg gazebo)"/>
   <arg name="load_robot_description" value="true"   unless = "$(arg gazebo)"/>
 </include>


<node if = "$(arg gazebo)" pkg="rosservice" type="rosservice" name="unpause_gazebo" args="call --wait /gazebo/unpause_physics" />

 <!-- the controller does not manage the hand, use a joint_state_publisher to handle it -->
<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
  <param name="/use_gui" value="false"/>
  <param name="rate" value="500"/>
  <rosparam param="/source_list">[/nj_220_2_7/joint_states]</rosparam>
</node>

  <!-- in high_dof_6_moveit_config package the joint_states is remapped to joint_states_desired -->
<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />

<!-- load moveit -->
<!-- <include file="$(find nj_220_2_7_moveit_config)/launch/move_group.launch">
  <arg name="allow_trajectory_execution" value="true"/>
  <arg name="fake_execution_type" value="false"/>
  <arg name="info" value="true"/>
  <arg name="debug" value="$(arg debug)"/>
  <arg name="pipeline" value="$(arg pipeline)"/>
</include> -->


<node name="joint_target_publisher" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui">
  <param name="rate" value="500"/>
  <remap from="joint_states" to="/nj_220_2_7/joint_target"/>
</node>
<!-- load rviz -->
<include file="$(find nj_220_2_7_configurations)/launch/rviz.launch" if="$(arg use_rviz)">
  <arg name="rviz_config" value="$(find nj_220_2_7_configurations)/launch/control.rviz"/>
  <arg name="debug" value="$(arg debug)"/>
</include>

</launch>
